packages/wallet-sdk/src/connection/WalletLinkConnection.ts

```diff
@@ -88,120 +74,112 @@ export class WalletLinkConnection {
     this.diagnostic = diagnostic;
     this.listener = listener;
 
-    const ws = new WalletLinkWebSocket(`${linkAPIUrl}/rpc`, WebSocketClass);
-    ws.setConnectionStateListener(async (state) => {
-      // attempt to reconnect every 5 seconds when disconnected
-      this.diagnostic?.log(EVENTS.CONNECTED_STATE_CHANGE, {
-        state,
-        sessionIdHash: Session.hash(session.id),
+    this.ws = new WalletLinkWebSocket({
+      linkAPIUrl,
+      session,
+      WebSocketClass,
+      listener: this,
+    });
```

Comment:
`linkAPIUrl => ws url` conversion has been migrated to `WalletLinkWebSocket` class

![image](https://github.com/coinbase/coinbase-wallet-sdk/assets/966662/c00ce868-e23b-47ed-a4de-eab028f2e975)